bin/lib/installation.py

```diff
@@ -115,4 +118,7 @@ def installers_for(install_context, nodes, enabled):
         if target_type not in _INSTALLER_TYPES:
             raise RuntimeError(f"Unknown installer type {target_type}")
         installer_type = _INSTALLER_TYPES[target_type]
-        yield installer_type(install_context, target)
+        try:
+            yield installer_type(install_context, target)
+        except Exception as e:
```

Comment:
Please don't catch `Exception` here - that's far too general and will mean even a typo in the python code (SyntaxError) will be just logged as a warning. I know this is done elsewhere in the codebase, but `RuntimeError` is probably what you want here.

And I will take an action to fix up the other `Exception`s elsewhere